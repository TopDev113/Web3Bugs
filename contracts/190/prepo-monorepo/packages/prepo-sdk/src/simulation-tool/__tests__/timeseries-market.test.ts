import Actor from "../actor";
import TimeSeriesMarket from "../timeseries-market";

describe("there is some activity on a timeseries market", () => {
  let market: TimeSeriesMarket;
  beforeAll(() => {
    const usdMinted = 10000;
    const lp = new Actor("lp", usdMinted);
    const tradeSize = 1000;
    const trader = new Actor("trader", usdMinted);
    const marketConfig = {
      bounds: { floor: 0.2, ceil: 0.8 },
      fee: 0.02,
      protocolFee: 0.01,
    };
    market = new TimeSeriesMarket("hello", marketConfig, usdMinted, lp);
    const allInLp = new Actor("allInLp", 1000);
    // market.mint(allInLp, 1000);
    // market.depositLiquidityNoLeftover(allInLp);

    market.openPosition(trader, "long", tradeSize);
    market.checkpoint(lp, allInLp);
  });

  test("timeseries can be exported", () => {
    const timeSeries = market.getTimeSeries();
    expect(timeSeries).toMatchSnapshot();
  });
});
