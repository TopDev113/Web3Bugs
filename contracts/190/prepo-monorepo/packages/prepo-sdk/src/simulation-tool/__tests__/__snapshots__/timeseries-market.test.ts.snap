// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`there is some activity on a timeseries market timeseries can be exported 1`] = `
Array [
  Object {
    "allInLpPercentProfit": 0,
    "leftoverLpPercentProfit": -0.015190355663386982,
    "longPoolValue": 5107.590854922462,
    "longPrice": 0.5508006469209438,
    "maxLongTradeValueWithoutSignificantSlippage": 88.65390096720918,
    "maxShortTradeValueWithoutSignificantSlippage": 80.06123321315539,
    "shortPoolValue": 4740.505588443668,
    "shortPrice": 0.4492036660383646,
    "time": 1609459200000,
  },
]
`;
